.github/workflows/merge-to-main.yml

```
@@ -0,0 +1,85 @@
+          git merge-tree $(git merge-base origin/main origin/dev) origin/main origin/dev | grep -q "^@@" && {
+            echo "❌ Merge conflicts detected. Please resolve conflicts manually."
+            exit 1
+          } || {
+            echo "✅ No merge conflicts detected"
+          }
+
+      - name: Create Pull Request
+        id: create_pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:8:32: Double quote to prevent globbing and word splitting

(shellcheck)